Update UI and Styling for Contact Management App

mawalou14 opened the pull request:
Closes: #823

- **Title:** Update UI and Styling for Contact Management App
- **Name:** Mouhamadou Awalou
- **Idenitfy yourself:** Contributor in Hacktoberfest

### Describe the add-ons or changes you've made 📃
In this contribution, I have updated the UI and styling of the Contact Management App located in the advanced section of the Angular-js projects. The changes include improved visual elements for better user experience, including a responsive layout, updated color schemes, and enhanced component styling.
Also,  updated the readme.md file of this project by replacing the screenshots with the new look.


### Screenshots 📷
![contact management app](https://github.com/user-attachments/assets/a051d1f6-20e2-4565-aeb6-ab11d1a8ad2d)
![noContactFound](https://github.com/user-attachments/assets/0ee5a53b-7d8f-4131-a324-3680d5fc4fc1)
![successPopup](https://github.com/user-attachments/assets/6a066b01-11e4-4981-aefd-c443e12210fa)




### Type of change ☑️
What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


### Checklist: ☑️
- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CONTRIBUTING.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.


### Note to reviewers 📄
Please review the UI changes and provide feedback on the overall design and usability improvements. I appreciate your input!. The background image used is a free picture downloaded from freepik.com.

Avdhesh-Varshney reviewed (changes requested):
Delete all the `.angular` cache files and other any un-necessary files.

Avdhesh-Varshney commented:
Can you tell what you have improved except bg addition?

mawalou14 commented:
> Can you tell what you have improved except bg addition?

I've made serceral improvements beyond just adding the background image. Here are the changes and improvements I made:

- [x] Added toastr notifications for all operations ( Add, Update, Delete).
- [x] Updated the subscription syntax to follow best practices.
- [x]  Removed unnecessary full-page reloads by notifying the contact list component via the contact service, ensuring smooth updates.
- [x]  Added meaningful comments to help other developers understand the changes.
- [x]  Improved UI layout by making the header fixed and placing the search bar in the header.
- [x] Added a "No contact found" message when the contact list is empty.
- [x] Updated the README.md for better documentation and understanding.

mawalou14 commented:
> Delete all the `.angular` cache files and other any un-necessary files. 

Hi @Avdhesh-Varshney. Done, I've pushed the requested changes, and all feedback has been implemented.

Avdhesh-Varshney commented on Angular-JS-Projects/Advanced/Contact-Manager/screenshot.webp:
Undo this image. `screenshot.webp` file is mandatory in every project.

mawalou14 replied:
Hi, @Avdhesh-Varshney, Done I restored the mandatory screenshot.webp file

Avdhesh-Varshney commented on Angular-JS-Projects/Advanced/Contact-Manager/README.md:
No screenshot file. No image is there to render in readme file?
![image](https://github.com/user-attachments/assets/7397d0d4-3b01-40f1-9f88-2d0f92d42769)

Avdhesh-Varshney replied:
You should also review the readme file and show me also

mawalou14 replied:
Alright, no worries

mawalou14 replied:
> You should also review the readme file and show me also

Hi, @Avdhesh-Varshney, the screenshots (05 in number) have been updated, then added all those to the readme file for inorder to ease the understanding of the project UI. 
So now the readme file is updated and well documented. Below is the a screenshot of part of the readme file.

![Screenshot 2024-10-16 223615](https://github.com/user-attachments/assets/e437a098-28d5-42d9-8cd5-b80a3d02de71)
